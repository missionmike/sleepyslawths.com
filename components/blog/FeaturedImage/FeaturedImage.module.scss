@import 'styles/mixins';

.featuredImageContainer {
  position: relative;
  overflow: hidden;
  width: 50%;
  min-height: 500px;

  @include media-breakpoint-down(lg) {
    width: calc(100% - 2rem);
    margin: 0 auto;
    min-height: 200px;
  }
}
