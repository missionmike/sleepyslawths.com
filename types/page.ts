export type PageSEODef = 'common' | 'blog' | 'tag';
