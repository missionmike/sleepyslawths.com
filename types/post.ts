export interface Post {
  content?: string;
  path?: string;
  href?: string;
  frontMatter?: { [key: string]: string };
}
