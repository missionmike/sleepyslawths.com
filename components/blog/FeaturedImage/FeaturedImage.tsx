import { Col } from 'react-bootstrap';
import Image from 'next/image';
import Link from 'next/link';
import { Post } from 'types/post';
import React from 'react';
import styles from './FeaturedImage.module.scss';

const FeaturedImage = ({ post = {}, priority = false }: { post: Post; priority?: boolean }) => {
  const { frontMatter, href } = post;

  return (
    <Col className={styles.featuredImageContainer}>
      <Link href={href}>
        <Image
          src={`/static/images/${frontMatter.featuredImage}`}
          fill
          sizes="(max-width: 768px) 100vw,
(max-width: 1200px) 50vw,
33vw"
          alt=""
          style={{ objectFit: 'cover' }}
          priority={priority}
        ></Image>
      </Link>
    </Col>
  );
};

export { FeaturedImage };
