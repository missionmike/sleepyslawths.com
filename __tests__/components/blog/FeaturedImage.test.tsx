import { FeaturedImage } from 'components/blog/FeaturedImage/FeaturedImage';
import { Post } from 'types/post';
import React from 'react';
import { render } from '@testing-library/react';

describe('FeaturedImage component', () => {
  it('renders the component', () => {
    const post: Post = {
      href: '#',
      frontMatter: {
        featuredImage: 'javascript:;',
      },
    };
    const { container } = render(<FeaturedImage post={post} />);
    expect(container).toMatchSnapshot();
  });

  it('renders the component with priority', () => {
    const post: Post = {
      href: '#',
      frontMatter: {
        featuredImage: 'javascript:;',
      },
    };
    const { container } = render(<FeaturedImage post={post} priority={true} />);
    expect(container).toMatchSnapshot();
  });
});
