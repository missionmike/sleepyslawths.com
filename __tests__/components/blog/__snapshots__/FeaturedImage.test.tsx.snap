// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedImage component renders the component 1`] = `
<div>
  <div
    class="featuredImageContainer col"
  >
    <a
      href="#"
    >
      <img
        alt=""
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="(max-width: 768px) 100vw,
(max-width: 1200px) 50vw,
33vw"
        src="/_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=3840&q=75"
        srcset="/_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=256&q=75 256w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=384&q=75 384w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=640&q=75 640w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=750&q=75 750w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=828&q=75 828w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=1080&q=75 1080w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=1200&q=75 1200w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=1920&q=75 1920w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=2048&q=75 2048w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </a>
  </div>
</div>
`;

exports[`FeaturedImage component renders the component with priority 1`] = `
<div>
  <div
    class="featuredImageContainer col"
  >
    <a
      href="#"
    >
      <img
        alt=""
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="(max-width: 768px) 100vw,
(max-width: 1200px) 50vw,
33vw"
        src="/_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=3840&q=75"
        srcset="/_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=256&q=75 256w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=384&q=75 384w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=640&q=75 640w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=750&q=75 750w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=828&q=75 828w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=1080&q=75 1080w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=1200&q=75 1200w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=1920&q=75 1920w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=2048&q=75 2048w, /_next/image?url=%2Fstatic%2Fimages%2Fjavascript%3A%3B&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </a>
  </div>
</div>
`;
